src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -355,5 +355,210 @@ int SoftwareFallback(ulong v)
+        //    value &= ~mask;
+
+        //    return btr;
+        //}
+
+        ///// <summary>
+        ///// Clears the specified bit in a mask and returns whether it was originally set.
+        ///// Similar in behavior to the x86 instruction BTR.
+        ///// </summary>
+        ///// <param name="value">The value.</param>
+        ///// <param name="bitOffset">The ordinal position of the bit to clear.
+        ///// Any value outside the range [0..31] is treated as congruent mod 32.</param>
+        //[MethodImpl(MethodImplOptions.AggressiveInlining)]
+        //public static bool ClearBit(ref int value, int bitOffset)
+        //{
+        //    int mask = 1 << bitOffset;
+
+        //    bool btr = (value & mask) != 0;
+        //    value &= ~mask;
+
+        //    return btr;
+        //}
+
+        /// <summary>
+        /// Clears the specified bit in a mask and returns the new value.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to clear.
+        /// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static byte ClearBit(byte value, int bitOffset)
+            => (byte)ClearBit((uint)value, bitOffset & 7);
+
+        /// <summary>
+        /// Clears the specified bit in a mask and returns the new value.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to clear.
+        /// Any value outside the range [0..31] is treated as congruent mod 32.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint ClearBit(uint value, int bitOffset)
+            => value & ~(1u << bitOffset);
+
+        /// <summary>
+        /// Clears the specified bit in a mask and returns the new value.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to clear.
+        /// Any value outside the range [0..31] is treated as congruent mod 32.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int ClearBit(int value, int bitOffset)
+            => value & ~(1 << bitOffset);
+
+        ///// <summary>
+        ///// Sets the specified bit in a mask and returns whether it was originally set.
+        ///// Similar in behavior to the x86 instruction BTS.
+        ///// </summary>
+        ///// <param name="value">The value.</param>
+        ///// <param name="bitOffset">The ordinal position of the bit to write.
+        ///// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        //[MethodImpl(MethodImplOptions.AggressiveInlining)]
+        //public static bool InsertBit(ref byte value, int bitOffset) // TODO: offset should maybe be uint
```

Comment:
This style of `InsertBit` more matches `BTS` more [closely](https://www.felixcloutier.com/x86/bts).
Not sure if it should be surfaced or not (it is _not_ used by any known callers)